src/ffi/ceval.rs

```diff
@@ -1,7 +1,5 @@
-#[cfg(not(Py_LIMITED_API))]
-use crate::ffi::code::FreeFunc;
 use crate::ffi::object::PyObject;
-use crate::ffi::pystate::{PyThreadState, Py_tracefunc};
+use crate::ffi::pystate::PyThreadState;
 use std::os::raw::{c_char, c_int, c_void};
 
 extern "C" {
```

Comment:
Can't select the exact line, but PyEval_CallObjectWithKeywords is marked deprecated in 3.9. However, if I mark it with `#[cfg_attr(Py_3_9, deprecated(note = "Python 3.9"))]`, I can't compile due to `deny(warnings)` on this crate (it's used by `PyEval_CallObject` below).

Maybe we could consider being a bit more specific with the crate's `deny` attribute? https://github.com/rust-unofficial/patterns/blob/master/anti_patterns/deny-warnings.md

Reply:
Thanks, I recently added this and I agree it's now an antipattern.

I've opened #1340 to remove the attribute.

As specifically here, I think we will need to put `#[allow(deprecated)]` on the call inside `PyEval_CallObject` so that this passes CI. Mildly annoying, but hey!

Reply:
This is fine, I think. I think the only gotcha is that we need to add the deprecation to `PyEval_CallObject` as well.